src/indra_cogex/client/queries.py

```diff
@@ -1116,12 +1141,11 @@ def get_stmts_for_stmt_hashes(
             evidence_map=evidence_map,
             evidence_limit=evidence_limit,
         )
-
     if not return_evidence_counts:
         return rv
     evidence_counts = {
-        stmt.get_hash(): rel.data["evidence_count"] for rel, stmt in zip(rels, stmts)
-    }
+        stmt.get_hash():  len(stmt.evidence) for stmt in rv}
+
```

Comment:
Can you explain this change? What about the previous implementation using `rels` wasn't working?

Reply:
The reason for asking this is because the value of `len(stmt.evidence)` would depend on the number of evidences we add to the statements in `enrich_statements`, and we don't always load all statements in there, so it's better to look at the evidence count that's already present in the relationship data, which is what we look at when using `rels`.

Reply:
I made this change because the rel.data["evidence_count"] consistently gave wrong evidence count values. 
I think we had a meeting a few weeks ago where you directly checked the evidence counts of statement hashes (we did this because for the GO Curation page, the sorting wasnt' happening according to the evidence counts). 

We found this : 
ipdb> for row in formatted_stmts: print(row[2], row[4], evidence_counts[int(row[2][1:-1])])
"13301055625711548" 1 6
"33745117863160054" 2 6
"-895900855983881" 1 4
"-19674726551250739" 1 3
"29599780063516948" 1 3
"22101717839556401" 5 3
"9663687963398921" 2 2
"14104567373107142" 1 2
"5902341912423440" 1 2
"11445082854558021" 1 2
"10896500623179539" 2 2
"12971441607560370" 1 1
"-491020425052903" 2 1
"23851336085223248" 2 1
"6315846869510301" 3 1
"-5415505964213929" 1 1
"-12964199264109371" 2 1
"-25887716695418031" 1 1
"-2723778492011540" 4 1
"-16633675098857024" 2 1
"21429626957792552" 1 1
"31004054634342170" 2 1
"-1126040494860724" 1 1
"-17498188144827063" 5 1
"27488399416633972" 1 1
The middle column seems to be correct by checking the number of evidences available for that statement hash in the graph database, while the count in the right column coming from evidence_counts seems to be incorrect.  

We found that evidence counts coming directly from the statement objects is more accurate since it reflects the actual number of evidence objects attached to each statement.

Reply:
I don't think this change is the right solution to that problem though, we'll end up with other problems down the line if we rely `len(stmt.evidence)` instead of what comes out of `rels`.

Reply:
Resolved by ccb666dd49e828c79a3782a22aa9d4277c08396f and e175597081476b6a33acb97b11bb831ae8c9017c